app/src/main/java/io/gnosis/safe/ui/settings/owner/OwnerSeedPhraseViewModel.kt

```diff
@@ -47,8 +49,11 @@ class OwnerSeedPhraseViewModel
             safeLaunch {
                 updateState { ImportOwnerKeyState.Error(InvalidPrivateKey) }
             }
+        } else if (!keyCanBeUsedForSigning(input.hexAsBigInteger())) {
+            safeLaunch {
+                updateState { ImportOwnerKeyState.Error(InvalidPrivateKey) }
+            }
         } else {
-
             safeLaunch {
```

Comment:
it would be more legible if this method has just one top safeLaunch block instead of launching them for every if branch